redis/crc.py

```diff
@@ -0,0 +1,28 @@
+from binascii import crc_hqx
+
+# Redis Cluster's key space is divided into 16384 slots.
+# For more information see: https://github.com/redis/redis/issues/2576
+REDIS_CLUSTER_HASH_SLOTS = 16384
+
+__all__ = [
+    "crc16",
+    "key_slot",
+    "REDIS_CLUSTER_HASH_SLOTS"
+]
+
+
+def crc16(data):
+    return crc_hqx(data, 0)
+
+
+def key_slot(key, bucket=REDIS_CLUSTER_HASH_SLOTS):
+    """Calculate key slot for a given key.
+    :param key - bytes
+    :param bucket - int
+    """
+    start = key.find(b"{")
+    if start > -1:
+        end = key.find(b"}", start + 1)
+        if end > -1 and end != start + 1:
+            key = key[start + 1: end]
+    return crc16(key) % bucket
```

Comment:
Since crc16(key) isn't used elsewhere in the codebase, this could change?

```
return crc_hqx(data,0) % bucket
```

Reply:
Maybe be kind to the future reader and comment a link to: https://redis.io/topics/cluster-spec

Specifically the piece in the doc: 'HASH_SLOT = CRC16(key) mod 16384' is probably of interest.